src/cargo/core/compiler/fingerprint/dep_info.rs

```diff
@@ -129,6 +152,16 @@ impl EncodedDepInfo {
         }
         return Some(EncodedDepInfo { files, env });
 
+        fn read_version(bytes: &mut &[u8]) -> Option<u8> {
+            let _size = read_usize(bytes)?;
+            let path_type = read_u8(bytes)?;
+            if path_type != u8::MAX {
+                // Old depinfo. Give up parsing it.
+                return None;
+            }
```

Comment:
imo this function deserves comments, even if its pointing back to `EncodedDepInfo`s comment